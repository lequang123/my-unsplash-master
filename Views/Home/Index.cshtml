@{
    ViewData["Title"] = "Home Page";
}

@section Styles {
    <environment include="Development">
        <link rel="stylesheet" href="~/css/dev/pages/home/index.chunk.css" />
    </environment>
    <environment exclude="Development">
        <link rel="stylesheet" href="~/css/pro/pages/home/index.chunk.css" asp-append-version="true"/>
    </environment>
}
@section Scripts{
    <environment include="Development">
        <script src="~/js/dev/pages/home/index.chunk.js"></script>
    </environment>
    <environment exclude="Development">
        <script src="~/js/pro/pages/home/index.chunk.js" asp-append-version="true"></script>
    </environment>
}

<div id="app-container"></div>