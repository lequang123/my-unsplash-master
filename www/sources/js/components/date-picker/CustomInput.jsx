import React, { Fragment } from 'react';
import { Form } from 'react-bootstrap';

export default class CustomInput extends React.PureComponent {
    constructor(props) {
        super(props);
    }

    render() {
        const { disabled, onClick, value } = this.props;

        return (
            <Fragment>
                <Form.Control as='input' readOnly disabled={disabled} onClick={onClick} value={value} />
                <i className='date-picker-icon fa fa-calendar' onClick={onClick} />
            </Fragment>
        );
    }
}