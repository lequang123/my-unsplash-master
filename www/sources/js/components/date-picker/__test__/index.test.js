import React from 'react';
import Index from '../index';
import DatePicker from 'react-datepicker';
import CustomInput from '../CustomInput';
import { shallow } from 'enzyme';

describe('index', () => {
    it('should have correct snapshots', () => {
        // Action
        const index = shallow(<Index />);

        // Assert
        expect(index).toMatchSnapshot();
    });

    describe('render', () => {
        it('should return DatePicker', () => {
            // Arrange
            const props = {
                className: 'className',
                showTimeSelect: true
            };

            // Action
            const index = shallow(<Index {...props} />);

            // Assert
            expect(index.props().className).toBe(`form-control form-control-sm datetime-picker ${props.className}`);
            expect(index.props().dateFormat).toBe('MMM dd, yyyy hh:mm aa');
            expect(index.props().timeFormat).toBe('hh:mm aa');
            expect(index.props().timeIntervals).toEqual(5);
            expect(index.props().customInput).toEqual(<CustomInput />);
        });
        
        it('should return DatePicker with showTimeSelect false', () => {
            // Arrange
            const props = {
                showTimeSelect: false
            };

            // Action
            const index = shallow(<Index {...props} />);

            // Assert
            expect(index.props().dateFormat).toBe('MMM dd, yyyy');
        });
    });
});