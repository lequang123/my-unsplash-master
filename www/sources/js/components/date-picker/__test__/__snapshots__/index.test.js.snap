// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index should have correct snapshots 1`] = `ShallowWrapper {}`;
